entrypoint.sh

```diff
@@ -1,40 +1,64 @@
 #!/bin/sh
 #
-# Sync/Diff rules to a Cortex cluster using the cortex tool
+# Interact with the Cortex Ruler API using the cortextool
 
 err() {
   echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: " + "$@" >&2
 }
 
+# For commands that interact with the server, we need to verify that the
+# CORTEX_TENANT_ID and CORTEX_ADDRESS are set.
+verifyTenantAndAddress() {
+  if [ -z "${CORTEX_TENANT_ID}" ]; then
+    err "CORTEX_TENANT_ID has not been set."
+    exit 1
+  fi
+
+  if [ -z "${CORTEX_ADDRESS}" ]; then
+    err "CORTEX_ADDRESS has not been set."
+    exit 1
+  fi
+}
+
+LINT_CMD=lint
+CHECK_CMD=check
+PREPARE_CMD=prepare
+SYNC_CMD=sync
+DIFF_CMD=diff
 
 if [ -z "${RULES_DIR}" ]; then
+  echo "RULES_DIR not set, using './' as a default."
   RULES_DIR="./"
 fi
 
 if [ -z "${ACTION}" ]; then
-  ACTION="diff"
-fi
-
-if [ -z "${CORTEX_TENANT_ID}" ]; then
-  err "CORTEX_TENANT_ID has not been set."
+  err "ACTION has not been set."
   exit 1
 fi
 
-if [ -z "${CORTEX_ADDRESS}" ]; then
-  err "CORTEX_ADDRESS has not been set."
-  exit 1
-fi
-
-
 case "${ACTION}" in
-  sync)
+  $SYNC_CMD)
+    verifyTenantAndAddress
     OUTPUT=$(/usr/bin/cortextool rules sync --rule-dirs="${RULES_DIR}")
     STATUS=$?
     ;;
-  diff)
+  $DIFF_CMD)
+    verifyTenantAndAddress
     OUTPUT=$(/usr/bin/cortextool rules diff --rule-dirs="${RULES_DIR}")
     STATUS=$?
     ;;
+  $LINT_CMD)
+    OUTPUT=$(/usr/bin/cortextool rules lint --rule-dirs="${RULES_DIR}")
+    STATUS=$?
+    ;;
+  $PREPARE_CMD)
+    OUTPUT=$(/usr/bin/cortextool rules prepare -i --rule-dirs="${RULES_DIR}")
```

Comment:
We're using `-i` here to modify in-place. I have created #3 to follow up on this at some point I _think_ we should be using the GitHub action `inputs` API to receive params for this action instead of using environment variables.